Fix parse boolean value error with elasticsearch

acris-cp opened the pull request:
### 1. Why is this change necessary?
If I assign a dynamic product group to a category and assign the filter "active = 1" or a other boolean filter to the dynamic product group, an error "failed to create query: Can't parse boolean value [1], expected [true] or [false]" occures on searching. A fallback to the normal database search will happen - it shouldn't.

### 2. What does this change do, exactly?
It converts the values of boolean fields to boolean values before add it to the elasticsearch search query to prevent searching for values like 1 or 0 for boolean fields.

### 3. Describe each step to reproduce the issue or behaviour.
1. Create a dynamic product group
2. Add a filter there like "active": "Yes"
3. Assign the dynamic product group to be showed in a category listing.
4. Load the category in storefront - no error occures but a fallback search will happen to search in the normal database
5. If we print the error in Elasticsearch/Framework/DataAbstractionLayer/ElasticsearchEntitySearcher.php - line 90, the following error message will occur: 
{"error":{"root_cause":[{"type":"query_shard_exception","reason":"failed to create query: Can't parse boolean value [1], expected [true] or [false]","index_uuid":"QOA9DtcnTQGCt1pErCTvfA","index":"sw_cp6__product_6adfee4faa9c40369f7babd0c283bcb7_1604165948"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"sw_cp6__product_6adfee4faa9c40369f7babd0c283bcb7_1604165948","node":"6CVO0lrvS-qTYxt_QsPWZA","reason":{"type":"query_shard_exception","reason":"failed to create query: Can't parse boolean value [1], expected [true] or [false]","index_uuid":"QOA9DtcnTQGCt1pErCTvfA","index":"sw_cp6__product_6adfee4faa9c40369f7babd0c283bcb7_1604165948","caused_by":{"type":"illegal_argument_exception","reason":"Can't parse boolean value [1], expected [true] or [false]"}}}]},"status":400}

### 4. Please link to the relevant issues (if any).
https://issues.shopware.com/issues/NEXT-11784

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

shyim commented:
Fixed with https://github.com/shopware/shopware/commit/40ec126c20d2b2e7177acfc3569aa1e62ffa9efd

acris-cp committed:
Fix parse boolean value error with elasticsearch

If I assign a dynamic product group to a category and assign the filter "active = 1" or a other boolean filter to the dynamic product group, an error "failed to create query: Can't parse boolean value [1], expected [true] or [false]" occures on searching. A fallback to the normal database search will happen - it shouldn't.

soebbing commented:
We also see this error - a LOT!

bneumann97 commented:
Hey @acris-cp, thanks for your contribution 💙. Unfortunately you forgot the [changelog](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) file and a test would also be nice.

bneumann97 commented:
This PR was closed due to inactivity. If this change is still important to you, feel free to create a new pull request.
For more information about our contribution guidelines, see https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline
